src/App.tsx

```diff
@@ -1,10 +1,47 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { getTodos } from './api/todos';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { Notification } from './Components/Notification';
 
-const USER_ID = 0;
+const USER_ID = 7011;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedStatus, setSelectedStatus] = useState<Status>(Status.All);
+  const [isError, setIsError] = useState(false);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (selectedStatus) {
+        case Status.Active:
+          return !todo.completed;
+        case Status.Completed:
+          return todo.completed;
+        default:
+          return todo;
+      }
+    });
+  }, [selectedStatus, todos]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => {
+        setIsError(true);
+        setTimeout(() => {
+          setIsError(false);
+        }, 3000);
+      });
+  }, [todos]);
```

Comment:
You're adding 'todos' as a dependency to the useEffect, which could lead to infinite re-fetching if the state is updated. Consider removing 'todos' from the dependency array (i.e fetch it only in the first render). 🔄